src/fetch/headers.js

```diff
@@ -118,11 +118,9 @@ class Headers {
     }
   }
 
-  * keys() {
-    // return Object.keys(this[INTERNALS].map).sort();
-    for (const name of Object.keys(this[INTERNALS].map).sort()) {
-      yield name;
-    }
+  keys() {
+    return Array.from(this[INTERNALS].map.keys())
+      .sort();
   }
 
   * values() {
```

Comment:
why not just `return this[INTERNALS].map.values()` ?

Reply:
Same here: because `Map` is insertion ordered and the values are expected ordered by keys. Have a look at the `node-fetch` test cases or try it out in the browser.

Reply:
ah, I don't know the spec says they have to be sorted.

Reply:
the spec isn't crystal clear in this regard but `node-fetch`, Chrome and Safari do return the headers in lexicographical order.